C/common/datapoint_utility.cpp

```diff
@@ -0,0 +1,224 @@
+*/
+DatapointUtility::Datapoints *DatapointUtility::findListElement(Datapoints *dict, const string& key) {
+	return findDictOrListElement(dict, key, DatapointValue::T_DP_LIST);
+}
+
+/**
+ * Search a list or dictionary from a key
+ *
+ * @param dict : parent dictionary
+ * @param key : key to research
+ * @param type : type of data searched
+ * @return vector of datapoint otherwise null pointer
+*/
+DatapointUtility::Datapoints *DatapointUtility::findDictOrListElement(Datapoints *dict, const string& key, DatapointValue::dataTagType type) {
+	Datapoint *dp = findDatapointElement(dict, key);
+	
+	if (dp == nullptr) {
+		return nullptr;
+	}
+
+	DatapointValue& data = dp->getData();
+	if (data.getType() == type) {
+		return data.getDpVec();
+	}
+	
+	return nullptr;
+}
+
+/**
+ * Search a DatapointValue from a key
+ *
+ * @param dict : parent dictionary
+ * @param key : key to research 
+ * @return corresponding datapointValue otherwise null pointer
+*/
+DatapointValue *DatapointUtility::findValueElement(Datapoints *dict, const string& key) {
+	
+	Datapoint *dp = findDatapointElement(dict, key);
+	
+	if (dp == nullptr) {
+		return nullptr;
+	}
+
+	return &dp->getData();
+}
+
+/**
+ * Search a Datapoint from a key
+ *
+ * @param dict : parent dictionary
+ * @param key : key to research
+ * @return corresponding datapoint otherwise null pointer
+*/
+Datapoint *DatapointUtility::findDatapointElement(Datapoints *dict, const string& key) {
+	if (dict == nullptr) {
+		return nullptr;
+	}
+	
+	for (Datapoint *dp : *dict) {
+		if (dp->getName() == key) {
+			return dp;
+		}
+	}
+	return nullptr;
+}
+
+/**
+ * Search a string from a key
+ *
+ * @param dict : parent dictionary
+ * @param key : key to research
+ * @return correponding string otherwise empty string
+*/
+string DatapointUtility::findStringElement(Datapoints *dict, const string& key) {
+	
+	Datapoint *dp = findDatapointElement(dict, key);
+	
+	if (dp == nullptr) {
+		return "";
+	}
+
+	DatapointValue& data = dp->getData();
+	const DatapointValue::dataTagType dType(data.getType());
+	if (dType == DatapointValue::T_STRING) {
+		return data.toStringValue();
+	}
+
+	return "";
+}
+
+/**
+ * Method to delete elements in a vector
+ * 
+ * @param dps dict of values 
+ * @param key key of dict 
+*/
+void DatapointUtility::deleteValue(Datapoints *dps, const string& key) {
+	Datapoints::iterator it1 = dps->end();
+	Datapoint *d = nullptr;
+	for (Datapoints::iterator it = dps->begin(); it != dps->end(); it++){
+		if ((*it)->getName() == key) {
+			it1 = it;
+			d = *it;
+			break;
+		}
+	}
+
+	if (d != nullptr) {
+		dps->erase(it1);
+		delete d;
+	}
+}
+
+/**
+ * Generate default attribute integer on Datapoint
+ * 
+ * @param dps dict of values 
+ * @param key key of dict
+ * @param valueDefault value attribute of dict
+ * @return pointer of the created datapoint
+ */
+Datapoint *DatapointUtility::createIntegerElement(Datapoints *dps, const string& key, long valueDefault) {
+
+	deleteValue(dps, key);
```

Comment:
Ok with that, but this method implies that the insertion gets a `O(n)` complexity.

Reply:
I don't understand, can you give me more details?

Reply:
I Just mean that unconditionnaly calling delete will imply a complete loop over all elements, which leads to a complexity of `O(n)`. That may lead to performances issues. however, I don't see any real option as nothing is designed to avoid this linear complexity...

Reply:
Ah ok. thanks for the details. 
When calling the delete method, the loop will browse the elements until it finds the one with the key. It doesn't go through all the elements.
If not, I have no other idea.